Support non-decorated async coroutine in fork (#1255)
cocotb/cocotb: documentation/source/coroutines.rst

is determined by which type of function it appears in, not by the
sub-coroutine being called.

> .. note::

Reviewer: ```suggestion
.. versionchanged:: 1.4
```

    The :any:`cocotb.coroutine` decorator is no longer necessary for ``async def`` functions. Such functions can be

>     used un-decorated wherever decorated coroutines are accepted, including :keyword:`yield` statements and

Reviewer 2: ```suggestion
    used without the ``@cocotb.coroutine`` decorator wherever decorated coroutines are accepted, including :keyword:`yield` statements and
```

    :any:`cocotb.fork`.

>     .. versionadded:: 1.3

Reviewer: ```suggestion
```

to make up for it being at the top of this section.
